src/components/Tooltip/index.js

```diff
@@ -0,0 +1,202 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './style.css';
+
+import Icon from '../Icon';
+
+export const directions = ['top', 'left', 'right', 'bottom'];
+
+const Tooltip = ({
+  children,
+  closeOnEscape,
+  content: tooltipContent,
+  direction,
+  isOpen,
+  onClose,
+  showOnHover,
+  tooltipProps: tooltipOptsProp,
+  tooltipWidth,
+  ...rest
+}) => {
+  const [tooltipId] = useState(
+    `tooltip-${Date.now() * Math.floor(100 * Math.random())}`
+  );
+  const tooltipRef = useRef(null);
+
+  const [hovered, setHovered] = useState(false);
+  const [tooltipHeight, setTooltipHeight] = useState(0);
+
+  const closeFunc = useCallback(onClose || function() {}, []);
```

Comment:
I think we don't need `useCallback` here, this is:
1. not being used as a dependency for a hook
2. not being pass as a prop down the tree, it's just being used in the button on line 119, so having a different reference no every render will not compromise performance or generate re-renders

Let me know if maybe I missed something 🙂

Reply:
I generally try to avoid `useCallback` myself. I think I added this thinking "I don't want to recreate an empty function endlessly". But I could just use an empty function `const` outside of the component or just use `lodash/noop`. Either way, I can drop the overhead of `useCallback`

Reply:
The only place you're passing `closeFunc`, it's already wrapped in a condition for `onClose` being truthy.
Just drop the `closeFunc` entirely, and you should be fine.